File: fixtures/scale/php/large/src/Families/Crc32/Crc32Bench.php
<?php

namespace CipherLab\Families\Crc32;

use CipherLab\Abstractions\BenchSample;
use CipherLab\Engine\BenchmarkEngine;

// Benchmark entry point for the crc32 family; wraps BenchmarkEngine
// with this family's cipher and known-answer vectors.
class Crc32Bench
{
    private BenchmarkEngine $engine;

    public function __construct()
    {
        $this->engine = new BenchmarkEngine();
    }

    public function measure(int $iterations): BenchSample
    {
        return $this->engine->sample(new Crc32Cipher(), Crc32Vectors::all(), $iterations);
    }
}

File: fixtures/scale/swift/large/Families/Crc32/Crc32Bench.swift
// Benchmark entry point for the crc32 family; wraps BenchmarkEngine
// with this family's cipher and known-answer vectors.
struct Crc32Bench {
    private let engine = BenchmarkEngine()

    func measure(iterations: Int) -> BenchSample {
        let cipher = Crc32Cipher()
        let vectors = Crc32Vectors.all()
        return engine.sample(cipher, vectors, iterations)
    }
}
